trace/trace_opentracing_test.go

```diff
@@ -0,0 +1,135 @@
+package trace
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/thundra-io/thundra-lambda-agent-go/test"
+	"github.com/thundra-io/thundra-lambda-agent-go/thundra"
+)
+
+const (
+	mainDuration = 100
+	f1Duration   = 50
+	f1Name       = "f1"
+	f2Duration   = 30
+	f2Name       = "f2"
+)
+
+// EXAMPLE HANDLERS
+func handler1(ctx context.Context, s string) (string, error) {
+	span, ctx := opentracing.StartSpanFromContext(ctx, "test-operation")
+	defer span.Finish()
+
+	span.SetTag("tagKey", "tagValue")
+	time.Sleep(time.Millisecond * mainDuration)
+	return fmt.Sprintf("Happy monitoring with %s!", s), nil
+}
+
+func handler2(ctx context.Context, s string) (string, error) {
+	span, ctx := opentracing.StartSpanFromContext(ctx, "test-operation")
+	defer span.Finish()
+
+	span.SetTag("tagKey", "tagValue")
+
+	f := func(ctx context.Context, operationName string, duration time.Duration) {
+		data, ctx := opentracing.StartSpanFromContext(ctx, operationName)
+		defer data.Finish()
+		time.Sleep(time.Millisecond * duration)
+	}
+	f(ctx, "f1", f1Duration)
+	f(ctx, "f2", f2Duration)
+
+	time.Sleep(time.Millisecond * mainDuration)
+	return fmt.Sprintf("Happy monitoring with %s!", s), nil
+}
+
+func TestSpanTransformation(t *testing.T) {
```

Comment:
Can you add another test case for collecting spans in case of timeout?

Reply:
Can you also add another test for collecting spans when there are multiple goroutines?